src/test/java/com/atlassian/bitbucket/jenkins/internal/fixture/JenkinsProjectRule.java

```diff
@@ -0,0 +1,173 @@
+    private Job activeJob;
+    private ProjectType activeJobType;
+    // The list of all jobs that have been created as part of testing this job
+    private List<Job> instantiatedJobs = new ArrayList<>();
+
+    private JenkinsProjectRule(Builder builder) {
+        this.jenkinsRule = builder.jenkinsRule;
+        this.testedProjectTypes = builder.projectTypes;
+    }
+
+    @Override
+    public Statement apply(Statement base, Description description) {
+        return new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                // If the method was not annotated with a project initialization, run as normal
+                if (testedProjectTypes.isEmpty()) {
+                    base.evaluate();
+                } else {
+                    try {
+                        initializeProjects();
+                        for (ProjectType projectType : testedProjectTypes) {
+                            setActiveJob(projectType);
+                            base.evaluate();
+                        }
+                    } finally {
+                        cleanupProjects();
+                    }
+                }
+            }
+        };
+    }
+
+    /**
+     * Returns the job that the project rule is currently testing.
+     *
+     * @return the currently active job
+     */
+    public Job getJob() {
+        return activeJob;
+    }
+
+    /**
+     * Sets the SCM of the currently active job
+     *
+     * @param scm the SCM to add to the currently active job
+     */
+    public void setSCM(SCM scm) throws IOException {
+        switch(activeJobType) {
+            case FREESTYLE:
+                ((FreeStyleProject) activeJob).setScm(scm);
+                break;
+            case PIPELINE:
+                ((WorkflowJob) activeJob).setDefinition(new CpsScmFlowDefinition(scm, "Jenkinsfile"));
+                break;
+            case MULTIBRANCH:
+                // TODO: Add Multibranch Support
+                break;
+        }
+    }
+
+    /**
+     * Gets the SCM of the currently active job
+     *
+     * @return the SCM of the currently active job
+     */
+    @Nullable
+    public SCM getSCM() {
+        switch(activeJobType) {
+            case FREESTYLE:
+                return ((FreeStyleProject) activeJob).getScm();
+            case PIPELINE:
+                return ((CpsScmFlowDefinition) ((WorkflowJob) activeJob).getDefinition()).getScm();
+            case MULTIBRANCH:
+                // TODO: Add Multibranch Support
+                return null;
+            default:
+                return null;
+        }
+    }
+
+    private void setActiveJob(ProjectType type) {
+        activeJob = instantiatedJobs.stream()
+                .filter(job -> job.getClass() == type.getJobClass())
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException("Project of type " + type.toString() + " not found."));
+        activeJobType = type;
```

Comment:
Instead of maintaining a separate `activeJobType` (with the slight increased chance of missing updating it somewhere), can we always derive it from `activeJob` on-demand?